### ISSUE TITLE
Additional signal handler to reload/reexec configuration

### ISSUE THREAD

**Issue by lahwaacz** [lefcha/imapfilter#73]
When using the `enter_idle()` method and the system suspends/hibernates, after resume imapfilter is still in idle even if some messages arrived.

A solution is to manually restart imapfilter, but it is necessary to fill in password again. It could also be solved automatically if imapfilter had a callback to handle e.g. SIGUSR1 to reload/reexec the configuration.

**Comment by lenucksi** [comment]
Do I understand this ticket right that this is not intended to reload the configuration file after manual changes to it, i.e. the addition of another filter, using some signal such as SIGUSR1?

Currently I always restart imapfilter using some script after I have added or changed a rule. Would it be possible to provide the behaviour as illustrated above?

**Comment by lefcha** [comment]
This issue revealed a couple of problems, one was being that imapfilter didn't have a way to reload its config when the user sent as signal to it, and another one that imapfilter recovers a connection and automatically reenters IDLE mode, after the computer was waken up after suspend.

The first problem is a bit difficult to tackle, but I have provided a simple workaround in one of my posts, ie. to use 2 config files, where the main config file executes the commands on the auxiliary config file every few minutes, but also whenever there has been a manual change to the auxiliary config file.  This way one can modify what commands are executed without stopping and starting imapfilter.

The second problem was solved by introducing a new `reenter` option, which makes imapfilter skip reentering IDLE mode when a lost connection is recovered.  This is useful when a user has a some filering commands in the beginning and then imapfilter goes into IDLE mode, and all this is inside a loop.  So next time a connection is lost and recovered, imapfilter will continue after the IDLE mode command, and thus go in the beginning of the loop and start all over.
